syft/workers/base.py

```diff
@@ -970,8 +1012,13 @@ def search(self, query: Union[List[Union[str, int]], str, int]) -> List[PointerT
                 # set garbage_collect_data to False because if we're searching
                 # for a tensor we don't own, then it's probably someone else's
                 # decision to decide when to delete the tensor.
-                ptr = obj.create_pointer(garbage_collect_data=False, owner=sy.local_worker).wrap()
-                results.append(ptr)
+                if obj.owner is not self:
```

Comment:
If the worker is the owner of the object - then simply return the object

Reply:
This plays a bit fast and loose with types (tensors vs pointers), which works fine in Python but becomes kind of a pain to encode in the Syft protocol and deal with in other languages. If this is a convenience change that isn't required for the main purpose of the PR, I'd probably rather leave part alone.

Reply:
This PR does not depend necessary on this (I can call ```get``` in the ```load function```).
I think the problem was more "semantics" wise - like if you have the tensor  - why you should get a pointer to that tensor and not the effective tensor?

Reply:
To make the return type consistent. If sometimes I get one thing and sometimes I get another thing, then I have to check what was returned in order to know how to process it, which proliferates `if` statements through the code base.

Reply:
Yep, will update this.